src/main/java/tr/com/srdc/cda2fhir/validation/ValidatorImpl.java

```diff
@@ -25,51 +25,76 @@
 import java.io.FileReader;
 import java.io.IOException;
 import java.io.OutputStream;
-
+import java.util.ArrayList;
 import java.util.List;
 
 import org.apache.commons.io.IOUtils;
 import org.hl7.fhir.dstu3.model.Bundle;
 import org.hl7.fhir.dstu3.model.Bundle.BundleEntryComponent;
 import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import ca.uhn.fhir.context.FhirContext;
-
 import ca.uhn.fhir.validation.FhirValidator;
 import ca.uhn.fhir.validation.IValidatorModule;
 import ca.uhn.fhir.validation.SchemaBaseValidator;
-import ca.uhn.fhir.validation.ValidationResult;
 import ca.uhn.fhir.validation.SingleValidationMessage;
-
+import ca.uhn.fhir.validation.ValidationResult;
 import ca.uhn.fhir.validation.schematron.SchematronBaseValidator;
 
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
 public class ValidatorImpl implements IValidator {
 
 	private final Logger logger = LoggerFactory.getLogger(ValidatorImpl.class);
-	
+
 	/**
 	 * Constructs a validator using the default configuration.
 	 */
 	public ValidatorImpl() {
 
+	}
+
+	@Override
+	public List<ValidationResult> validateBundleResultsOnly(Bundle bundle) {
+		if (bundle == null) {
+			logger.warn("The bundle to be validated is null. Returning null.");
+			return null;
+		}
+		if (bundle.getEntry().isEmpty()) {
+			logger.warn("The bundle to be validated is empty. Returning null");
+			return null;
+		}
+
+		logger.info("Validating the bundle containing " + bundle.getEntry().size() + " entries");
+
+		List<ValidationResult> results = new ArrayList<ValidationResult>();
+
+		// traverse the entries of the bundle
+		for (BundleEntryComponent entry : bundle.getEntry()) {
+			if (entry != null && entry.getResource() != null) {
+				results.add(validateResourceResultOnly(entry.getResource()));
+			} else {
+				logger.warn("An entry of the bundle validator was running on was found null. Ignoring the entry.");
```

Comment:
can you just rephrase this error where it shows up? It's just bad english. "Bundle validator encountered a null entry, ignoring." would be better.

Reply:
Will do. That's what was already in the validator when we got it.